stl/inc/yvals_core.h

```diff
@@ -470,6 +470,12 @@
 #define _CONSTEXPR20 inline
 #endif // ^^^ inline (not constexpr) in C++17 and earlier ^^^
 
+#if 0 // TRANSITION: #ifdef __cpp_constexpr_dynamic_alloc

```

Comment:
Shouldn't `#if __cpp_constexpr_dynamic_alloc` be used instead of `#ifdef`?

Reply:
Feature-test macros are never `0`. If feature is not supported they are undefined. Examples:
https://github.com/microsoft/STL/blob/fffbd8fe05cd1a21adf2284eb6acef69e982b3c7/stl/inc/yvals_core.h#L1068-L1070
https://github.com/microsoft/STL/blob/fffbd8fe05cd1a21adf2284eb6acef69e982b3c7/stl/inc/yvals_core.h#L319-L325
https://github.com/microsoft/STL/blob/fffbd8fe05cd1a21adf2284eb6acef69e982b3c7/stl/inc/compare#L107-L115

Reply:
Yes, but undefined macro are replaced with `0`/`false`. And [SD-6](https://isocpp.org/std/standing-documents/sd-6-sg10-feature-test-recommendations) sugests to use such form

Reply:
Filed #517. The question is not specific for this PR